### ISSUE TITLE
feat: Add `form` property to IonButton to submit from outside form

### ISSUE THREAD

**Issue by TamoMaez** [ionic-team/ionic-framework#21194]
# Bug Report

**Ionic version:**
**4.x**

**Current behavior:**
IonButton does not trigger submission of the form. You need this when your submit button is outside the form itself.
```
<form id="some-form">....</form>
<ion-button form="some-form">...</ion-button>
```

**Expected behavior:**
Should trigger the submit event of the form like so: https://www.w3schools.com/tags/att_button_form.asp

**Steps to reproduce:**

Add the code below:
```
<form id="some-form" (ngSubmit)="submit('Triggered')"></form>
<ion-button form="some-form">Trigger submit</ion-button>
```

Tap on the button and nothing will happen

**Related code:**

https://stackblitz.com/edit/ionic-v4-angular-tabs-qx5u9v?file=src%2Fapp%2Ftab1%2Ftab1.page.html

**Workaround:**
Hide native button element and trigger it by reference
```
...<form ... id="transaction-form">...</form>
...
<ion-footer>
    <ion-toolbar button (click)="submitBtn.click()">
        <ion-title>Some title</ion-title>
        <button hidden #submitBtn type="submit" form="transaction-form"></button>
    </ion-toolbar>
</ion-footer>
```

**Comment by indraraj26** [comment]
if you place ion-button with type submit inside form it should trigger automatically. let me know if this doesn't work for you

**Comment by liamdebeasi** [comment]
@indraraj26 Is correct. You need `type="submit"` on the `ion-button` otherwise it will not submit the form.

**Comment by TamoMaez** [comment]
> @indraraj26 Is correct. You need `type="submit"` on the `ion-button` otherwise it will not submit the form.

Hi @indraraj26, adding `type="submit" ` doesn't fix the problem https://stackblitz.com/edit/ionic-v4-angular-tabs-qx5u9v?file=src%2Fapp%2Ftab1%2Ftab1.page.html

When you are outside the form, adding the form attribute of the ion-button (which exists for this purpose) doesn't trigger submission. Please reopen

**Comment by indraraj26** [comment]
Hi @liamdebeasi , 

He is just assigned the id of form to button, Native button is triggering the ngSubmit but not ion-button which is wrapper around button.

```
<form id="some-form" (ngSubmit)="submit()"></form>
    <ion-button type="submit" form="some-form">Trigger form submit (doesn't work)</ion-button>
    <hr />
    <button type="submit" form="some-form">Trigger form submit (does work)</button>
```

If possible please re-open

**Comment by liamdebeasi** [comment]
Ok looks like this is only an issue when used outside the `form` as stated.

I am marking this as a feature request as `ion-button` does not support the `form` property. If anyone would like to make a PR for this feature, please let me know!

There are 2 things we need to do:

1. Add the `form` property to `ion-button`.
2. Update https://github.com/ionic-team/ionic/blob/master/core/src/components/button/button.tsx#L155 to select the form element if the `form` property is provided.

**Comment by indraraj26** [comment]
Hi @liamdebeasi , I cloned ionic repo yesterday i had so many errors like cleancss not recognized etc that i resolved but now i dont know how to include ionic-angular to my project.

**Comment by TakumaKira** [comment]
Hi @indraraj26,

I think [this](https://github.com/ionic-team/ionic-framework/blob/master/angular/README.md#testing-ng-add-in-ionic) will help.

**Comment by itsLucario** [comment]
Hi @liamdebeasi ,

I have created PR which addresses the feature request and enables form submission even outside the form with added `form` property.

**Comment by tanetrail** [comment]
Hi

Is this feature available in Ionic-React or when will it be available in ionic-react. not able to submit outside of the form. Thanks for all the help

**Comment by stewones** [comment]
this still an issue as of sep 2021

I'm having to fake a submit button inside the form as workaround

**Comment by darron1217** [comment]
Really simple thing.. any updates?

**Comment by sean-perkins** [comment]
Hello everyone 👋 we have had some great contributions in the past towards this PR. After reviewing the proposed changes, I want to outline a few requirements so that if anyone is still interested, they can submit a PR and it can be reviewed quickly. 

The proposed introduction for this feature is the following property:
```ts
@Prop() form?: string | HTMLFormElement;
```

It should accept a `string` of the form owner's `id`. It will query the closest form with that id, if defined. Alternatively, it accepts a reference directly to the form element. This is useful for framework implementations that support working with refs.

Tests are a must for these changes. We need to validate with Playwright, that the form is submitted with both use cases (by form id and by reference). 

Thanks!

**Comment by postnerd** [comment]
@sean-perkins I've created a PR yesterday (after I was facing a similar challenge with an app I am working on). I tried to follow your instructions and also added support for referencing a HTMLFormElement. But since this is my first PR for this project, I'm not quite sure if my solution fits your ideas.

Please let me know if I should add/change something.

**Comment by sean-perkins** [comment]
This feature was completed in https://github.com/ionic-team/ionic-framework/pull/25913 and will be available in Ionic 6.3.0 (upcoming release).
